pkg/bg/bg.go

```diff
@@ -0,0 +1,64 @@
+// Package bg provides primitives for the background tasks
+package bg
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/rs/zerolog"
+)
+
+type config struct {
+	name   string
+	logger *zerolog.Logger
+}
+
+type Opt func(*config)
+
+func WithName(name string) Opt {
+	return func(cfg *config) { cfg.name = name }
+}
+
+func WithLogger(logger zerolog.Logger) Opt {
+	return func(cfg *config) { cfg.logger = &logger }
```

Comment:
not sure if we really need a pointer `*zerolog.Logger` type if it's already passed by value `logger zerolog.Logger`.

Reply:
I used pointer to determine whether logger is set for this bg job. afaik there's no `logger.IsNop()`

Reply:
Does it panic when using a `Nop` logger to print logs?

Reply:
https://github.com/zeta-chain/node/pull/2428/commits/55626c964cf8850e36026aeca127059aa9821aa5